src/LoadOnline.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace AntoineFr\Online;
+
+use Carbon\Carbon;
+use Flarum\Api\Controller\ShowForumController;
+use Flarum\Settings\SettingsRepositoryInterface;
+use Flarum\User\User;
+use Psr\Http\Message\ServerRequestInterface;
+use Tobscure\JsonApi\Document;
+
+class LoadOnline
+{
+    protected $settings;
+
+    public function __construct(SettingsRepositoryInterface $settings)
+    {
+        $this->settings = $settings;
+    }
+
+    public function __invoke(ShowForumController $controller, &$data, ServerRequestInterface $request, Document $document)
+    {
+        $actor = $request->getAttribute('actor');
+        $ago = Carbon::now()->subtract(5, 'minutes');
+
+        $online = User::query()->whereRaw('? < last_seen_at', [$ago])->get();
+
+        $filtered = collect($online)->map(function ($user) use ($actor) {
+            if ($user->getPreference('discloseOnline') || $actor->can('viewLastSeenAt', $user)) {
+                return $user;
+            }
+        });
```

Comment:
I would suggest the following code because in my tests I had a `null` in the resulting array if the user chose to hide his online status.
```php
        $filtered = collect($online)->filter(function ($user) use ($actor) {
            return ($user->getPreference('discloseOnline') || $actor->can('viewLastSeenAt', $user));
        });
```

Reply:
Yeah right, i forget to [reject](https://laravel.com/docs/8.x/collections#method-reject) empty user.